nemo/lightning/pytorch/strategies/megatron_strategy.py

```diff
@@ -267,6 +267,8 @@ def __init__(
         elif ddp == "pytorch":
             self.ddp_config = None
             self.no_ddp_communication_hook = False
+        elif ddp is None:
+            self.ddp_config = None
```

Comment:
1. @marcromeyn are you OK with this change i.e. making ddp optional? (btw. let's also update typing in MegatronStrategy.__init__ params accordingly)
2. Maybe ddp = "pytorch" will do (the previous elif)?

Reply:
2. Seems like it could also work, as long as the optimiser can be left [uninitialised](https://github.com/NVIDIA/NeMo/pull/11257/files#diff-a4c5ebd530c68c3a7a6d243c283bc3c9a79f8021caba1266382841bcef8ee254L331-R340) (and [this](https://github.com/NVIDIA/NeMo/pull/11257/files#diff-10e32938a852d0a6ad12080bd3a9fcc8674ee6a22bea24b77c3f8cd8c371b3edR66))